### ISSUE TITLE
Force close with peer: Adding HTLC 9 too slow: killing connection

### ISSUE THREAD

**Issue by pira** [ElementsProject/lightning#5448]
### Issue and Steps to Reproduce
Running latest master build, dedicated server with no downtime, got a force closure with node 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f

### `getinfo` output
2022-07-21T11:42:45.322Z UNUSUAL 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-channeld-chan#263: Adding HTLC 9 too slow: killing connection
2022-07-21T11:42:45.323Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (9)
2022-07-21T11:44:41.165Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in CHANNELD_NORMAL: Disconnected
2022-07-21T11:44:43.550Z UNUSUAL 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer permanent failure in CHANNELD_NORMAL: channeld: received ERROR error channel 20efcd37b24836360df323bcd23c98fbf1159b0b3fc1077edbaf5e1d7383b9b2: bad reestablish revocation_number: 3 vs 136
2022-07-21T11:44:43.560Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: State changed from CHANNELD_NORMAL to AWAITING_UNILATERAL
2022-07-21T11:44:46.329Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:44:48.657Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected

...[cut 30 pages of the same error]...

2022-07-21T11:52:05.938Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:52:06.435Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
2022-07-21T11:52:07.930Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:52:08.442Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
2022-07-21T11:52:10.604Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:52:10.939Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
2022-07-21T11:52:12.820Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:52:12.930Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
2022-07-21T11:52:14.864Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:52:15.605Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
2022-07-21T11:52:17.497Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:52:17.821Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
2022-07-21T11:52:18.545Z UNUSUAL 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: Peer permanent failure in AWAITING_UNILATERAL: Funding transaction spent
2022-07-21T11:52:18.545Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: State changed from AWAITING_UNILATERAL to AWAITING_UNILATERAL
2022-07-21T11:52:18.546Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: State changed from AWAITING_UNILATERAL to FUNDING_SPEND_SEEN
2022-07-21T11:52:18.615Z INFO    0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-chan#263: State changed from FUNDING_SPEND_SEEN to ONCHAIN
2022-07-21T11:52:19.864Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
2022-07-21T11:52:22.497Z **BROKEN** 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close

**Comment by pira** [comment]
"version": "v0.11.0.1-343-g5abed48",

**Comment by rustyrussell** [comment]
OK, there are a few things going on here:

```
channeld: received ERROR error channel 20efcd37b24836360df323bcd23c98fbf1159b0b3fc1077edbaf5e1d7383b9b2: bad reestablish revocation_number: 3 vs 136
```

This is the fatal one: they close channel believing we sent crap.  **This was a bug in 0.11.0**, so I wonder what they're doing?

Second, we aggressively reconnect in AWAITING_UNILATERAL, which is kind of a waste, since we have nothing to say.

```
 Peer transient failure in AWAITING_UNILATERAL: Disconnected
2022-07-21T11:52:06.435Z BROKEN 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f-connectd: Peer did not close, forcing close
```

They stopped talking to us; fair enough, we're being annoying at this point.

**Comment by pira** [comment]
I unfortunately do not know who runs that node but he claims he runs "Core Lightning 0.11.2" (https://1ml.com/node/0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f)

**Comment by whitslack** [comment]
Here are the logs from the other side of this failure.

```
2022-07-21T11:42:25.882Z DEBUG   02…7c-channeld-chan#249444: peer_in WIRE_UPDATE_FEE
2022-07-21T11:42:25.889Z DEBUG   02…7c-channeld-chan#249444: update_fee 4580, range 253-57948
2022-07-21T11:42:26.842Z DEBUG   02…7c-channeld-chan#249444: We need 171087sat at feerate 4580 for 0 untrimmed htlcs: we have 16777215000msat/16777215000msat
2022-07-21T11:42:26.929Z DEBUG   02…7c-channeld-chan#249444: Setting LOCAL feerate to 4580
2022-07-21T11:42:26.960Z DEBUG   02…7c-channeld-chan#249444: peer updated fee to 4580
2022-07-21T11:42:26.987Z DEBUG   02…7c-channeld-chan#249444: peer_in WIRE_COMMITMENT_SIGNED
2022-07-21T11:42:27.122Z DEBUG   02…7c-channeld-chan#249444: Received commit
2022-07-21T11:42:27.159Z DEBUG   02…7c-channeld-chan#249444: Feerate: RCVD_ADD_HTLC->RCVD_ADD_COMMIT LOCAL now 4580
2022-07-21T11:42:27.278Z DEBUG   02…7c-channeld-chan#249444: Feerates are 4580/4581
2022-07-21T11:42:27.287Z DEBUG   02…7c-channeld-chan#249444: We need 171087sat at feerate 4580 for 0 untrimmed htlcs: we have 16777215000msat/16777215000msat
2022-07-21T11:42:27.304Z DEBUG   02…7c-channeld-chan#249444: Derived key 03…23 from basepoint 02…03, point 02…7f
2022-07-21T11:42:27.333Z DEBUG   02…7c-channeld-chan#249444: Received commit_sig with 0 htlc sigs
2022-07-21T11:42:46.200Z DEBUG   02…7c-channeld-chan#249444: Sending revoke_and_ack
2022-07-21T11:42:46.206Z DEBUG   02…7c-channeld-chan#249444: Feerate: RCVD_ADD_COMMIT->SENT_ADD_REVOCATION REMOTE now 4580
2022-07-21T11:42:46.220Z DEBUG   02…7c-chan#249444: attempting update blockheight 20…b2
2022-07-21T11:42:58.879Z DEBUG   02…7c-channeld-chan#249444: revoke_and_ack made pending: commit timer
2022-07-21T11:42:58.942Z DEBUG   02…7c-channeld-chan#249444: Sending master 1021
2022-07-21T11:42:59.082Z DEBUG   02…7c-chan#249444: got commitsig 135: feerate 4580, blockheight: 0, 0 added, 0 fulfilled, 0 failed, 0 changed
2022-07-21T11:42:59.255Z DEBUG   02…7c-channeld-chan#249444: ... , awaiting 1121
2022-07-21T11:42:59.363Z DEBUG   02…7c-channeld-chan#249444: Nope, got 1027 instead
2022-07-21T11:42:59.596Z DEBUG   02…7c-channeld-chan#249444: Got it!
2022-07-21T11:42:59.711Z DEBUG   02…7c-channeld-chan#249444: peer_out WIRE_REVOKE_AND_ACK
2022-07-21T11:42:59.809Z DEBUG   02…7c-channeld-chan#249444: Now dealing with deferred WIRE_CHANNELD_FEERATES
2022-07-21T11:42:59.900Z DEBUG   02…7c-channeld-chan#249444: Trying commit
2022-07-21T11:42:59.993Z DEBUG   02…7c-channeld-chan#249444: Feerate: SENT_ADD_REVOCATION->SENT_ADD_ACK_COMMIT REMOTE now 4580
2022-07-21T11:43:00.043Z DEBUG   02…7c-channeld-chan#249444: Creating commit_sig signature 135 30…01 for tx 02…20 wscript 52…ae key 02…cf
2022-07-21T11:43:00.066Z DEBUG   02…7c-channeld-chan#249444: Telling master we're about to commit...
2022-07-21T11:43:00.070Z DEBUG   02…7c-hsmd: Got WIRE_HSMD_VALIDATE_COMMITMENT_TX
2022-07-21T11:43:00.074Z DEBUG   02…7c-channeld-chan#249444: Sending master 1020
2022-07-21T11:43:00.225Z DEBUG   02…7c-channeld-chan#249444: ... , awaiting 1120
2022-07-21T11:43:00.238Z DEBUG   02…7c-channeld-chan#249444: Nope, got 1027 instead
2022-07-21T11:43:00.426Z DEBUG   02…7c-channeld-chan#249444: Got it!
2022-07-21T11:43:00.477Z DEBUG   02…7c-channeld-chan#249444: Sending commit_sig with 0 htlc sigs
2022-07-21T11:43:00.602Z DEBUG   02…7c-channeld-chan#249444: peer_out WIRE_COMMITMENT_SIGNED
2022-07-21T11:43:00.696Z DEBUG   02…7c-channeld-chan#249444: Now dealing with deferred WIRE_CHANNELD_FEERATES
2022-07-21T11:43:00.727Z DEBUG   02…7c-channeld-chan#249444: peer_in WIRE_UPDATE_ADD_HTLC
2022-07-21T11:43:00.736Z DEBUG   02…7c-channeld-chan#249444: NEW:: HTLC REMOTE 9 = RCVD_ADD_HTLC/SENT_ADD_HTLC 
2022-07-21T11:43:48.278Z INFO    02…7c-channeld-chan#249444: Peer connection lost
2022-07-21T11:43:48.278Z DEBUG   02…7c-channeld-chan#249444: Status closed, but not exited. Killing
2022-07-21T11:43:48.279Z INFO    02…7c-chan#249444: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (9)
2022-07-21T11:43:48.279Z DEBUG   02…7c-lightningd: Will try reconnect in 1 seconds
2022-07-21T11:43:48.616Z DEBUG   02…7c-connectd: Last ping unreturned: hanging up
2022-07-21T11:43:48.983Z DEBUG   02…7c-connectd: destroy_subd: 1 subds, to_peer conn (nil), read_to_die = 0
2022-07-21T11:43:49.307Z DEBUG   02…7c-connectd: disconnect
2022-07-21T11:43:49.675Z DEBUG   02…7c-connectd: peer_conn_closed
2022-07-21T11:43:49.930Z DEBUG   02…7c-lightningd: peer_disconnect_done
2022-07-21T11:43:49.933Z DEBUG   plugin-funder: Cleaning up inflights for peer id 02…7c
2022-07-21T11:43:50.056Z DEBUG   02…7c-connectd: processing now old peer gone
2022-07-21T11:43:50.150Z DEBUG   02…7c-chan#249444: Peer has reconnected, state CHANNELD_NORMAL: telling connectd to make active
2022-07-21T11:43:50.941Z DEBUG   02…7c-connectd: Handed peer, entering loop
2022-07-21T11:43:51.329Z DEBUG   02…7c-channeld-chan#249444: pid 10790, msgfd 887
2022-07-21T11:43:51.329Z DEBUG   02…7c-chan#249444: Already have funding locked in (and ready to announce)
2022-07-21T11:43:51.348Z DEBUG   02…7c-channeld-chan#249444: option_static_remotekey = 1, option_anchor_outputs = 0
2022-07-21T11:43:51.349Z DEBUG   02…7c-channeld-chan#249444: init REMOTE: remote_per_commit = 02…57, old_remote_per_commit = 03…a0 next_idx_local = 136 next_idx_remote = 136 revocations_received = 134 feerates { SENT_ADD_ACK_COMMIT:4580 RCVD_ADD_ACK_REVOCATION:4581 } range 253-57955 blockheights { RCVD_ADD_ACK_REVOCATION:0 }, our current 745881
2022-07-21T11:43:51.350Z DEBUG   02…7c-hsmd: Got WIRE_HSMD_GET_PER_COMMITMENT_POINT
2022-07-21T11:43:51.352Z DEBUG   02…7c-channeld-chan#249444: peer_out WIRE_CHANNEL_REESTABLISH
2022-07-21T11:43:51.352Z DEBUG   02…7c-hsmd: Got WIRE_HSMD_GET_PER_COMMITMENT_POINT
2022-07-21T11:43:51.353Z DEBUG   02…7c-channeld-chan#249444: billboard: Sent reestablish, waiting for theirs
2022-07-21T11:43:52.069Z DEBUG   02…7c-connectd: peer_in WIRE_GOSSIP_TIMESTAMP_FILTER
2022-07-21T11:43:58.931Z DEBUG   02…7c-channeld-chan#249444: peer_in WIRE_CHANNEL_REESTABLISH
2022-07-21T11:43:58.933Z DEBUG   02…7c-channeld-chan#249444: Got reestablish commit=4 revoke=3
2022-07-21T11:43:58.934Z DEBUG   02…7c-channeld-chan#249444: peer_out WIRE_ERROR
2022-07-21T11:43:58.935Z DEBUG   02…7c-channeld-chan#249444: billboard perm: bad reestablish revocation_number: 3 vs 136
2022-07-21T11:43:58.938Z UNUSUAL 02…7c-chan#249444: Peer permanent failure in CHANNELD_NORMAL: channeld: sent ERROR bad reestablish revocation_number: 3 vs 136
2022-07-21T11:43:58.943Z DEBUG   lightningd: sendrawtransaction: 02…20
2022-07-21T11:43:58.945Z INFO    02…7c-chan#249444: State changed from CHANNELD_NORMAL to AWAITING_UNILATERAL
2022-07-21T11:43:58.980Z DEBUG   plugin-bcli: sendrawtx exit 0 (bitcoin-cli -datadir=/var/lib/bitcoind sendrawtransaction 02…20) 
2022-07-21T11:43:59.950Z DEBUG   02…7c-connectd: disconnect
2022-07-21T11:44:21.819Z DEBUG   02…7c-connectd: destroy_subd: 1 subds, to_peer conn (nil), read_to_die = 1
2022-07-21T11:44:22.107Z DEBUG   02…7c-connectd: peer_conn_closed
2022-07-21T11:44:22.286Z DEBUG   02…7c-lightningd: peer_disconnect_done
2022-07-21T11:44:22.288Z DEBUG   plugin-funder: Cleaning up inflights for peer id 02…7c
2022-07-21T11:44:41.110Z DEBUG   02…7c-connectd: Connect IN
2022-07-21T11:44:41.331Z DEBUG   02…7c-connectd: peer_out WIRE_INIT
2022-07-21T11:44:42.019Z DEBUG   02…7c-connectd: peer_in WIRE_INIT
2022-07-21T11:44:42.158Z DEBUG   02…7c-chan#249444: Peer has reconnected, state AWAITING_UNILATERAL: telling connectd to make active
2022-07-21T11:44:42.360Z DEBUG   02…7c-connectd: Handed peer, entering loop
2022-07-21T11:44:43.094Z DEBUG   02…7c-lightningd: Telling connectd to send error 00…36
2022-07-21T11:44:43.383Z DEBUG   02…7c-connectd: destroy_subd: 1 subds, to_peer conn 0x557e20858f68, read_to_die = 0
2022-07-21T11:44:44.107Z DEBUG   02…7c-connectd: peer_out WIRE_ERROR
2022-07-21T11:44:44.253Z DEBUG   02…7c-connectd: peer_out WIRE_ERROR
2022-07-21T11:44:46.269Z DEBUG   02…7c-connectd: Connect IN
2022-07-21T11:44:46.359Z DEBUG   02…7c-connectd: peer_out WIRE_INIT
2022-07-21T11:44:47.217Z DEBUG   02…7c-connectd: peer_in WIRE_INIT
2022-07-21T11:44:47.304Z DEBUG   02…7c-connectd: reconnect
2022-07-21T11:44:47.388Z INFO    02…7c-chan#249444: Peer transient failure in AWAITING_UNILATERAL: Reconnected
2022-07-21T11:44:47.388Z DEBUG   02…7c-lightningd: Will try reconnect in 1 seconds
2022-07-21T11:44:48.629Z **BROKEN** 02…7c-connectd: Peer did not close, forcing close
2022-07-21T11:44:48.817Z DEBUG   02…7c-connectd: peer_conn_closed
2022-07-21T11:44:49.088Z DEBUG   02…7c-lightningd: peer_disconnect_done
2022-07-21T11:44:49.089Z DEBUG   plugin-funder: Cleaning up inflights for peer id 02…7c
2022-07-21T11:44:49.345Z DEBUG   02…7c-connectd: processing now old peer gone
2022-07-21T11:44:49.433Z DEBUG   02…7c-chan#249444: Peer has reconnected, state AWAITING_UNILATERAL: telling connectd to make active
2022-07-21T11:44:49.595Z DEBUG   02…7c-connectd: Handed peer, entering loop
2022-07-21T11:44:49.642Z DEBUG   02…7c-connectd: Connect IN
2022-07-21T11:44:49.690Z DEBUG   02…7c-connectd: peer_out WIRE_INIT
2022-07-21T11:44:50.346Z DEBUG   02…7c-connectd: peer_in WIRE_INIT
2022-07-21T11:44:50.378Z DEBUG   02…7c-connectd: reconnect
2022-07-21T11:44:50.424Z INFO    02…7c-chan#249444: Peer transient failure in AWAITING_UNILATERAL: Reconnected
2022-07-21T11:44:50.424Z DEBUG   02…7c-lightningd: Will try reconnect in 1 seconds
2022-07-21T11:44:50.581Z DEBUG   02…7c-lightningd: Telling connectd to send error 00…36
2022-07-21T11:44:50.714Z DEBUG   02…7c-connectd: destroy_subd: 1 subds, to_peer conn (nil), read_to_die = 0
2022-07-21T11:44:51.562Z DEBUG   02…7c-connectd: Connect IN
```

The last few lines (after `Connect IN`) kept repeating for quite some time until the commitment transaction eventually reached 3 confirmations on chain, after which time they ceased.

**Comment by rustyrussell** [comment]
OK, this is thorougly fixed in master (for v0.12) so I'm closing...

(Indeed, I stopped the annoying reconnect bombing too...)
